src/dmd/dsymbol.d

```diff
@@ -2316,14 +2316,18 @@ Dsymbol handleTagSymbols(ref Scope sc, Dsymbol s, Dsymbol s2, ScopeDsymbol sds)
         if (auto p = cast(void*)s2 in sc._module.tagSymTab)
         {
             Dsymbol s2tag = *p;
-            sd2 = s2tag.isStructDeclaration();
+            sd2 = s2tag.isScopeDsymbol();
             assert(sd2);        // only tags allowed in tag symbol table
         }
     }
 
     if (sd && sd2) // `s` is a tag, `sd2` is the same tag
     {
         if (log) printf(" tag is already defined\n");
+
+        if (sd.kind() != sd2.kind())  // being enum/struct/union must match
```

Comment:
Note, hidden string comparison.

Reply:
Not exactly, as D merges string duplicates.

Reply:
Only if they appear in the same compilation unit.